File: 1337-the-k-weakest-rows-in-a-matrix/1337-the-k-weakest-rows-in-a-matrix.java
import java.util.Vector;
class Solution {
    public int[] kWeakestRows(int[][] mat, int k) {
        PriorityQueue<int[]> pq = new PriorityQueue<>((a, b) -> {
            if (a[0] != b[0]) {
                return a[0] - b[0];
            } else {
                return a[1] - b[1];
            }
        });
        for(int i=0;i<mat.length;i++){
            int cnt=0;
            for(int j=0;j<mat[i].length;j++){
                 if(mat[i][j]==1) cnt++;
            }
            pq.offer(new int[]{cnt,i});
        }
        Vector<Integer> ans = new Vector<>();
        while(!pq.isEmpty() && k-->0){
            ans.add(pq.peek()[1]);
            pq.poll();
        }
        int result[] = new int[ans.size()];
        for(int i=0;i<ans.size();i++){
            result[i]=ans.get(i);
        }
        return result;
    }
}

File: 1337-the-k-weakest-rows-in-a-matrix/1337-the-k-weakest-rows-in-a-matrix.cpp
class Solution {
public:
    vector<int> kWeakestRows(vector<vector<int>>& mat, int k) {
        // {x,y}
        priority_queue<pair<int,int>,vector<pair<int,int>>,
        greater<pair<int,int>>> pq; //min heap
        for(int i=0;i<mat.size();i++){
            int cnt=0;//for each row that's why
            for(int j=0;j<mat[i].size();j++){
                if(mat[i][j] == 1) cnt++;
            }
            pq.push({cnt,i});
        }
        vector<int> ans;
        while(!pq.empty() and k-->0){
            ans.push_back(pq.top().second);
            pq.pop();
        }
        return ans;
    }
};